docs/how-it-works.md

```diff
@@ -0,0 +1,205 @@
+
+1. Takes your API description document,
+2. creates expectations based on requests and responses documented in the document,
+3. makes requests to tested API,
+4. checks whether API responses match the documented responses,
+5. reports the results.
+
+## Versioning
+
+Dredd follows [Semantic Versioning][]. To ensure certain stability of your Dredd installation (e.g. in CI), pin the version accordingly. You can also use release tags:
+
+- `npm install dredd` - Installs the latest published version including experimental pre-release versions.
+- `npm install dredd@stable` - Skips experimental pre-release versions. Recommended for CI installations.
+
+If the `User-Agent` header isn't overridden in the API description document, Dredd uses it for sending information about its version number along with every HTTP request it does.
+
+## Execution Life Cycle
+
+Following execution life cycle documentation should help you to understand how Dredd works internally and which action goes after which.
+
+1. Load and parse API description documents
+    - Report parse errors and warnings
+2. Pre-run API description check
+    - Missing example values for URI template parameters
+    - Required parameters present in URI
+    - Report non-parseable JSON bodies
+    - Report invalid URI parameters
+    - Report invalid URI templates
+3. Compile HTTP transactions from API description documents
+    - Inherit headers
+    - Inherit parameters
+    - Expand URI templates with parameters
+4. Load [hooks](hooks.md)
+5. Test run
+    - Report test run `start`
+    - Run `beforeAll` hooks
+    - For each compiled transaction:
+        - Report `test start`
+        - Run `beforeEach` hook
+        - Run `before` hook
+        - Send HTTP request
+        - Receive HTTP response
+        - Run `beforeEachValidation` hook
+        - Run `beforeValidation` hook
+        - [Perform validation](#automatic-expectations)
+        - Run `after` hook
+        - Run `afterEach` hook
+        - Report `test end` with result for in-progress reporting
+    - Run `afterAll` hooks
+6. Report test run `end` with result statistics
+
+## Limitations of API Description Formats
+
+## Automatic Expectations
+
+Dredd automatically generates expectations on HTTP responses based on examples in the API description with use of [Gavel.js][] library. Please refer to [Gavel][] rules if you want know more.
+
+### Response Headers Expectations
+
+- All headers specified in the API description must be present in the response.
+- Names of headers are validated in case-insensitive way.
+- Only values of headers significant for content negotiation are validated.
+- All other headers values can differ.
+
+When using [Swagger][], headers are taken from [`response.headers`][]. HTTP headers significant for content negotiation are inferred according to following rules:
+
+- [`produces`][] is propagated as response's `Content-Type` header.
+- Response's `Content-Type` header overrides any `produces`.
```

Comment:
yes, should be fixed or filed as an issue on Dredd/Swagger Adapter cc @pksunkara